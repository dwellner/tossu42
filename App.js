import React from "react";
import { Provider } from "react-redux";
import { createStore } from "redux";
import { persistStore, persistCombineReducers } from "redux-persist";
import storage from "redux-persist/lib/storage";
import { Platform, StatusBar, StyleSheet, View } from "react-native";
import { AppLoading, Asset, Font, Icon } from "expo";
import settingsReducer from "./reducers/SettingsReducer";
import AppNavigator from "./navigation/AppNavigator";

const persistanceConfig = {
  key: "root",
  storage
};

const rootReducer = persistCombineReducers(persistanceConfig, {
  settings: settingsReducer
});

const store = createStore(rootReducer);

export default class App extends React.Component {
  persistor = persistStore(store, {}, () =>
    this.setState({ isStoreHydrated: true })
  );

  state = {
    isStoreHydrated: false,
    isLoadingComplete: false
  };

  render() {
    if (
      !this.state.isLoadingComplete &&
      !this.state.isStoreHydrated &&
      !this.props.skipLoadingScreen
    ) {
      return (
        <AppLoading
          startAsync={this._loadResourcesAsync}
          onError={this._handleLoadingError}
          onFinish={this._handleFinishLoading}
        />
      );
    } else {
      return (
        <Provider store={store}>
          <View style={styles.container}>
            {Platform.OS === "ios" && <StatusBar barStyle="default" />}
            <AppNavigator />
          </View>
        </Provider>
      );
    }
  }

  _loadResourcesAsync = async () => {
    return Promise.all([
      Asset.loadAsync([
        require("./assets/images/robot-dev.png"),
        require("./assets/images/robot-prod.png")
      ]),
      Font.loadAsync({
        // This is the font that we are using for our tab bar
        ...Icon.Ionicons.font,
        // We include SpaceMono because we use it in HomeScreen.js. Feel free
        // to remove this if you are not using it in your app
        "space-mono": require("./assets/fonts/SpaceMono-Regular.ttf")
      })
    ]);
  };

  _handleLoadingError = error => {
    // In this case, you might want to report the error to your error
    // reporting service, for example Sentry
    console.warn(error);
  };

  _handleFinishLoading = () => {
    this.setState({ isLoadingComplete: true });
  };
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#333"
  }
});
