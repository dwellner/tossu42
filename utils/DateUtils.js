import moment from "moment";

export default {
  /** curent Date is used to open the UI to the current day by default.
   *  You should overwrite this value in automated tests to prevent changing dates in tests
   */
  currentDate: () => new Date().toISOString().substr(0, 10),
  nextDate: (date, delta) =>
    moment
      .utc(date)
      .add(delta, "day")
      .toISOString()
      .substr(0, 10),
  difference: (date1, date2) =>
    moment.utc(date1).diff(moment.utc(date2), "days")
};
