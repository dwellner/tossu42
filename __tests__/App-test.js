import 'react-native';
import React from 'react';
import App from '../App';
import renderer from 'react-test-renderer';
import NavigationTestUtils from 'react-navigation/NavigationTestUtils';
import DateUtils from "../utils/DateUtils"

describe('App snapshot', () => {
  jest.useFakeTimers();
  beforeEach(() => {
    DateUtils.currentDate = () => "2018-10-01"
    NavigationTestUtils.resetInternalState();
  });

  it('renders the loading screen', async () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the root without loading screen', async () => {
    const tree = renderer.create(<App skipLoadingScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
