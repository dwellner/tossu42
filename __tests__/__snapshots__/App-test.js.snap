// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot renders the loading screen 1`] = `null`;

exports[`App snapshot renders the root without loading screen 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#333",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "overflow": "hidden",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        collapsable={false}
        pointerEvents="auto"
        removeClippedSubviews={false}
        style={
          Array [
            Object {
              "flex": 1,
              "overflow": "hidden",
            },
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "opacity": 1,
              },
            ],
          ]
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            onLayout={[Function]}
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              onMoveShouldSetResponder={[Function]}
              onMoveShouldSetResponderCapture={[Function]}
              onResponderEnd={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderReject={[Function]}
              onResponderRelease={[Function]}
              onResponderStart={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              onStartShouldSetResponderCapture={[Function]}
              style={
                Array [
                  Object {
                    "flex": 1,
                    "flexDirection": "column-reverse",
                  },
                  Object {
                    "backgroundColor": "#000",
                  },
                ]
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <View
                  accessibilityElementsHidden={false}
                  pointerEvents="auto"
                  style={
                    Object {
                      "backgroundColor": "#E9E9EF",
                      "bottom": 0,
                      "left": 0,
                      "opacity": 1,
                      "paddingTop": 0,
                      "position": "absolute",
                      "right": 0,
                      "shadowColor": "black",
                      "shadowOffset": Object {
                        "height": 0,
                        "width": 0,
                      },
                      "shadowOpacity": 0.2,
                      "shadowRadius": 5,
                      "top": 0,
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                        Object {
                          "translateY": 0,
                        },
                      ],
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "height": "100%",
                        "width": "100%",
                      }
                    }
                  >
                    <Image
                      source={1}
                      style={
                        Array [
                          Object {
                            "bottom": 0,
                            "left": 0,
                            "position": "absolute",
                            "right": 0,
                            "top": 0,
                          },
                          Object {
                            "height": "100%",
                            "width": "100%",
                          },
                          undefined,
                        ]
                      }
                    />
                    <View
                      style={
                        Object {
                          "height": 190,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "alignItems": "stretch",
                            "backgroundColor": "#1119",
                            "borderBottomWidth": 1,
                            "borderColor": "#333",
                            "borderTopWidth": 1,
                            "flex": 1,
                            "marginBottom": 16,
                            "paddingBottom": 16,
                            "paddingTop": 30,
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "flex": 1,
                              "flexDirection": "row",
                            }
                          }
                        >
                          <View
                            style={
                              Object {
                                "flex": 1,
                              }
                            }
                          >
                            <View
                              accessibilityComponentType="button"
                              accessibilityTraits={
                                Array [
                                  "button",
                                ]
                              }
                              accessible={true}
                              isTVSelectable={true}
                              onResponderGrant={[Function]}
                              onResponderMove={[Function]}
                              onResponderRelease={[Function]}
                              onResponderTerminate={[Function]}
                              onResponderTerminationRequest={[Function]}
                              onStartShouldSetResponder={[Function]}
                              style={
                                Object {
                                  "opacity": 1,
                                }
                              }
                            >
                              <View
                                style={
                                  Array [
                                    Object {},
                                  ]
                                }
                              >
                                <Text
                                  accessible={true}
                                  allowFontScaling={true}
                                  ellipsizeMode="tail"
                                  style={
                                    Array [
                                      Object {
                                        "color": "#007AFF",
                                        "fontSize": 18,
                                        "padding": 8,
                                        "textAlign": "center",
                                      },
                                      Object {
                                        "color": "#333",
                                      },
                                    ]
                                  }
                                >
                                  ◀
                                </Text>
                              </View>
                            </View>
                          </View>
                          <View
                            style={
                              Object {
                                "alignItems": "center",
                                "flex": 4,
                              }
                            }
                          >
                            <Text
                              accessible={true}
                              allowFontScaling={true}
                              ellipsizeMode="tail"
                              style={
                                Object {
                                  "color": "#fff",
                                  "fontSize": 24,
                                }
                              }
                            >
                              Keskiviikko 3.10.
                            </Text>
                          </View>
                          <View
                            style={
                              Object {
                                "flex": 1,
                              }
                            }
                          >
                            <View
                              accessibilityComponentType="button"
                              accessibilityTraits={
                                Array [
                                  "button",
                                ]
                              }
                              accessible={true}
                              isTVSelectable={true}
                              onResponderGrant={[Function]}
                              onResponderMove={[Function]}
                              onResponderRelease={[Function]}
                              onResponderTerminate={[Function]}
                              onResponderTerminationRequest={[Function]}
                              onStartShouldSetResponder={[Function]}
                              style={
                                Object {
                                  "opacity": 1,
                                }
                              }
                            >
                              <View
                                style={
                                  Array [
                                    Object {},
                                  ]
                                }
                              >
                                <Text
                                  accessible={true}
                                  allowFontScaling={true}
                                  ellipsizeMode="tail"
                                  style={
                                    Array [
                                      Object {
                                        "color": "#007AFF",
                                        "fontSize": 18,
                                        "padding": 8,
                                        "textAlign": "center",
                                      },
                                      Object {
                                        "color": "#333",
                                      },
                                    ]
                                  }
                                >
                                  ▶
                                </Text>
                              </View>
                            </View>
                          </View>
                        </View>
                        <View
                          style={
                            Object {
                              "alignItems": "center",
                              "flex": 1,
                            }
                          }
                        >
                          <View
                            style={
                              Object {
                                "flex": 1,
                                "flexDirection": "row",
                              }
                            }
                          >
                            <Text
                              accessible={true}
                              allowFontScaling={true}
                              ellipsizeMode="tail"
                              style={
                                Object {
                                  "color": "#f442df",
                                  "fontSize": 20,
                                  "fontWeight": "bold",
                                }
                              }
                            >
                              26
                            </Text>
                            <Text
                              accessible={true}
                              allowFontScaling={true}
                              ellipsizeMode="tail"
                              style={
                                Object {
                                  "color": "#fff",
                                  "fontSize": 20,
                                  "marginLeft": 4,
                                }
                              }
                            >
                              päivää ennen
                            </Text>
                          </View>
                          <Text
                            accessible={true}
                            allowFontScaling={true}
                            ellipsizeMode="tail"
                            style={
                              Object {
                                "color": "#fff",
                                "fontSize": 20,
                              }
                            }
                          >
                            Helsinki City Maraton
                             - 
                            30.10.18
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
              <View
                onLayout={[Function]}
                pointerEvents="box-none"
                style={
                  Object {
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#FFF",
                      "transform": Array [
                        Object {
                          "translateX": -750,
                        },
                      ],
                    }
                  }
                >
                  <View
                    onLayout={[Function]}
                    pointerEvents="box-none"
                    style={
                      Object {
                        "backgroundColor": "#FFF",
                        "borderBottomColor": "#A7A7AA",
                        "borderBottomWidth": 0.5,
                        "height": 64,
                        "paddingBottom": 0,
                        "paddingLeft": 0,
                        "paddingRight": 0,
                        "paddingTop": 20,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    />
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      pointerEvents="box-none"
      style={
        Object {
          "backgroundColor": "#111",
          "borderTopColor": "rgba(0, 0, 0, .3)",
          "borderTopWidth": 0.5,
          "flexDirection": "row",
          "height": 49,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
        }
      }
    >
      <View
        accessibilityLabel="Päivänäkymä"
        accessible={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
            },
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "flexDirection": "column",
              "justifyContent": "flex-end",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "height": 29,
              },
              false,
              Object {
                "flex": 1,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "height": "100%",
                "justifyContent": "center",
                "minWidth": 25,
                "opacity": 1,
                "position": "absolute",
                "width": "100%",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#f442df",
                    "fontSize": 26,
                  },
                  Object {
                    "marginBottom": -3,
                  },
                  Object {
                    "fontFamily": "ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "height": "100%",
                "justifyContent": "center",
                "minWidth": 25,
                "opacity": 0,
                "position": "absolute",
                "width": "100%",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#ccc",
                    "fontSize": 26,
                  },
                  Object {
                    "marginBottom": -3,
                  },
                  Object {
                    "fontFamily": "ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
          </View>
        </View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "#fFF",
              "fontSize": 12,
              "marginBottom": 1.5,
              "textAlign": "center",
            }
          }
        >
          Päivänäkymä
        </Text>
      </View>
      <View
        accessibilityLabel="Ohjelma"
        accessible={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
            },
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "flexDirection": "column",
              "justifyContent": "flex-end",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "height": 29,
              },
              false,
              Object {
                "flex": 1,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "height": "100%",
                "justifyContent": "center",
                "minWidth": 25,
                "opacity": 0,
                "position": "absolute",
                "width": "100%",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#f442df",
                    "fontSize": 26,
                  },
                  Object {
                    "marginBottom": -3,
                  },
                  Object {
                    "fontFamily": "ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "height": "100%",
                "justifyContent": "center",
                "minWidth": 25,
                "opacity": 1,
                "position": "absolute",
                "width": "100%",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#ccc",
                    "fontSize": 26,
                  },
                  Object {
                    "marginBottom": -3,
                  },
                  Object {
                    "fontFamily": "ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
          </View>
        </View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "#929292",
              "fontSize": 12,
              "marginBottom": 1.5,
              "textAlign": "center",
            }
          }
        >
          Ohjelma
        </Text>
      </View>
      <View
        accessibilityLabel="Asetukset"
        accessible={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
            },
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "flexDirection": "column",
              "justifyContent": "flex-end",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "height": 29,
              },
              false,
              Object {
                "flex": 1,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "height": "100%",
                "justifyContent": "center",
                "minWidth": 25,
                "opacity": 0,
                "position": "absolute",
                "width": "100%",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#f442df",
                    "fontSize": 26,
                  },
                  Object {
                    "marginBottom": -3,
                  },
                  Object {
                    "fontFamily": "ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "height": "100%",
                "justifyContent": "center",
                "minWidth": 25,
                "opacity": 1,
                "position": "absolute",
                "width": "100%",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#ccc",
                    "fontSize": 26,
                  },
                  Object {
                    "marginBottom": -3,
                  },
                  Object {
                    "fontFamily": "ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
          </View>
        </View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "#929292",
              "fontSize": 12,
              "marginBottom": 1.5,
              "textAlign": "center",
            }
          }
        >
          Asetukset
        </Text>
      </View>
    </View>
  </View>
</View>
`;
